### PR TITLE
SITES-18799: Fetch AWS Cost Usage

### PR DESCRIPTION
**Pull request by sudo-buddy**
Please ensure your pull request adheres to the following guidelines:
- [ ] make sure to link the related issues in this description
- [ ] when merging / squashing, make sure the fixed issue references are visible in the commits, for easy compilation of release notes

## Related Issues

Thanks for contributing!

### CODE REVIEW

**Comment by solaris007** [comment]
you're missing `.releaserc.cjs` and you should commit `package-lock.json` as well.

File: packages/spacecat-shared-aws-api-client/package.json [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  },
+  "homepage": "https://github.com/adobe/spacecat-shared#readme",
+  "publishConfig": {
+    "access": "public"
+  },
+  "dependencies": {
+    "@adobe/fetch": "4.1.1",
+    "@adobe/helix-shared-wrap": "2.0.0",
+    "@adobe/helix-universal": "4.4.1",
+    "@aws-sdk/client-cost-explorer": "^3.496.0",
```
Comment (solaris007) [inline packages/spacecat-shared-aws-api-client/package.json:35]: please pin versions

File: packages/spacecat-shared-aws-api-client/package.json [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    "aws4": "1.12.0"
```
Comment (solaris007) [inline packages/spacecat-shared-aws-api-client/package.json:36]: what's this needed for? we should include only the aws libraries actually needed.

File: packages/spacecat-shared-aws-api-client/CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
+# [@adobe/spacecat-shared-example-v1.2.0](https://github.com/adobe/spacecat-shared/compare/@adobe/spacecat-shared-example-v1.1.0...@adobe/spacecat-shared-example-v1.2.0) (2024-01-12)
```
Comment (solaris007) [inline packages/spacecat-shared-aws-api-client/CHANGELOG.md:1]: the changelog should be empty for this package.

File: packages/spacecat-shared-aws-api-client/README.md [hunk 4]
```diff
@@ -0,0 +1,16 @@
+# Spacecat Shared - Example Wrapper
```
Comment (solaris007) [inline packages/spacecat-shared-aws-api-client/README.md:1]: please update the readme

File: packages/spacecat-shared-aws-api-client/src/index.d.ts [hunk 5]
```diff
@@ -0,0 +1,50 @@
+/*
```
Comment (solaris007) [inline packages/spacecat-shared-aws-api-client/src/index.d.ts:1]: please update the types

in license header use year 2024

File: packages/spacecat-shared-aws-api-client/src/index.js [hunk 6]
```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2023 Adobe. All rights reserved.
```
Comment (solaris007) [inline packages/spacecat-shared-aws-api-client/src/index.js:2]: 2024

File: packages/spacecat-shared-aws-api-client/package.json [hunk 7]
```diff
@@ -0,0 +1,38 @@
+{
+  "name": "@adobe/spacecat-shared-aws-api-client",
```
Comment (solaris007) [inline packages/spacecat-shared-aws-api-client/package.json:2]: are you building a generic AWS api client or just a cost-explorer? depending on the answer the package name needs adjusting.

if you're building a generic AWS api client, what would its features be?

File: packages/spacecat-shared-aws-api-client/package.json [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+    "url": "https://github.com/adobe/spacecat-shared/issues"
+  },
+  "homepage": "https://github.com/adobe/spacecat-shared#readme",
+  "publishConfig": {
+    "access": "public"
+  },
+  "dependencies": {
+    "@adobe/fetch": "4.1.1",
+    "@adobe/helix-shared-wrap": "2.0.0",
+    "@adobe/helix-universal": "4.4.1",
```
Comment (solaris007) [inline packages/spacecat-shared-aws-api-client/package.json:34]: these are lambda related, but you seem to be building a library. remove?

File: packages/spacecat-shared-aws-api-client/test/index.test.js [hunk 9]
```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2023 Adobe. All rights reserved.
```
Comment (solaris007) [inline packages/spacecat-shared-aws-api-client/test/index.test.js:2]: 2024

**Comment by sudo-buddy** [comment]
Raising a fresh new PR after renaming package name and the changes

**Pull request closed without merging**